[Disabled Tests] test/Error/rlexe.xml: tests tagged "fail" should be fixed and re-enabled #3118

tcare opened the pull request:
Re-enabled and updated prehistoric error source info baselines.

Stack information is such a luxury. :)

Fixes #3118

akroshg commented:
Would it be good idea to use unittestframework to assert for the error?

tcare commented:
These tests wanted to test source offsets which UnitTestFramework won't cover.

dilijev commented:
Maybe use https://github.com/Microsoft/ChakraCore/blob/master/test/UnitTestFramework/TrimStackTracePath.js to normalize the stack traces?

tcare commented:
Thanks for Doug for his input on what to do here.

I added the flag to make the errors more jshost-like, but throwing non-objects gives a different error in ch. In this case, it outputs the full path to the filename. I changed ch to just output the filename in this case, bringing it into line with similar situations in the surrounding code.

I also decided to have different baselines for Full and make it work like DebuggerCommon. Please see the corresponding ChakraFull review.

tcare commented:
In addition, I tried to make the non-object error look like other error messages, but context info like line numbers was not available. It seems we need to make more substantial changes to ch to bring this in line with jshost, but that is out of the scope of this change.

dilijev commented on bin/ch/WScriptJsrt.cpp:
```diff
@@ -1369,8 +1369,14 @@ bool WScriptJsrt::PrintException(LPCSTR fileName, JsErrorCode jsErrorCode)
                 {
                     const char *fName = fileName != nullptr ? fileName : "(unknown)";
+
+
```
nit: extra blank line
